lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -1774,13 +1766,19 @@ void HdVP2Mesh::_UpdateDrawItem(
                 // is MUCH faster to just not update the items we're not going to draw.
                 _delegate->GetVP2ResourceRegistry().EnqueueCommit(
                     [renderItem]() { renderItem->enable(false); });
-                return;
             } else {
                 stateToCommit._enabled = &drawItemData._enabled;
             }
         }
     }
 
+    // The above checks have captured the cases where the enabled state of the item
+    // changes. If the item is still not enabled, skip the rest of the update. Further
+    // code can cause the item to be disabled, but it can't change the item from enabled
+    // to disabled.
+    if (!drawItemData._enabled)
```

Comment:
This fixes an assertion I was sometimes seeing at line 2541. Sometime an render item that is hidden could have other changes and remain hidden. renderItemData._enable would match the new enable value and so the early exit would be missed.